internal/providers/pke/pkeworkflow/update_node_group_activity.go

```diff
@@ -0,0 +1,277 @@
+			"DesiredCapacity",
+			input.DesiredCapacity > 0,
+			fmt.Sprintf("%d", input.DesiredCapacity),
+		),
+		sdkCloudFormation.NewOptionalStackParameter(
+			"VolumeSize",
+			input.NodeVolumeSize > 0,
+			fmt.Sprintf("%d", input.NodeVolumeSize),
+		),
+		{
+			ParameterKey:     aws.String("ClusterName"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:     aws.String("NodeGroupName"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:     aws.String("VPCDefaultSecurityGroupId"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:     aws.String("ClusterSecurityGroup"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:     aws.String("VPCId"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:     aws.String("SSHLocation"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:     aws.String("SubnetIds"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:     aws.String("IamInstanceProfile"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:     aws.String("ClusterAutoscalerEnabled"),
+			UsePreviousValue: aws.Bool(true),
+		},
+		{
+			ParameterKey:   aws.String("PkeCommand"),
+			ParameterValue: aws.String(bootstrapCommand),
+		},
+		{
+			ParameterKey:     aws.String("PkeVersion"),
+			UsePreviousValue: aws.Bool(true),
+		},
+	}
+
+	// we don't reuse the creation time template, since it may have changed
+	updateStackInput := &cloudformation.UpdateStackInput{
+		ClientRequestToken: aws.String(sdkAmazon.NewNormalizedClientRequestToken(activity.GetInfo(ctx).WorkflowExecution.ID)),
+		StackName:          aws.String(input.StackName),
+		Capabilities:       []*string{aws.String(cloudformation.CapabilityCapabilityIam)},
+		Parameters:         stackParams,
+		Tags:               getNodePoolStackTags(input.ClusterName, input.ClusterTags),
+		TemplateBody:       aws.String(template),
+	}
+
+	_, err = cloudformationClient.UpdateStack(updateStackInput)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ValidationError" && strings.HasPrefix(awsErr.Message(), awsNoUpdatesError) {
+			return UpdateNodeGroupActivityOutput{}, nil
+		}
+
+		var awsErr awserr.Error
+		if errors.As(err, &awsErr) {
+			if awsErr.Code() == request.WaiterResourceNotReadyErrorCode {
+				err = pkgCloudFormation.NewAwsStackFailure(err, input.StackName, aws.StringValue(updateStackInput.ClientRequestToken), cloudformationClient)
+				err = errors.WrapIff(err, "waiting for %q CF stack create operation to complete failed", input.StackName)
+				if pkgCloudFormation.IsErrorFinal(err) {
+					return UpdateNodeGroupActivityOutput{}, cadence.NewCustomError(ErrReasonStackFailed, err.Error())
+				}
+				return UpdateNodeGroupActivityOutput{}, errors.WrapIff(err, "waiting for %q CF stack create operation to complete failed", input.StackName)
+			}
+		}
+	}
+
+	return UpdateNodeGroupActivityOutput{NodePoolChanged: true}, nil
```

Comment:
**Note**: not sure if it is intended, but currently in case no error occurs we always state the node pool has been changed which kind of defeats the purpose of including it in the output in my opinion unless we do it for interfacing purposes.

Reply:
Yeah, this was basically copied from the EKS UpdateNodeGroupActivity, which returns `NodePoolChanged{}, nil` in an if case a few lines above, so I would leave it as here as it is.

Reply:
Yeah, but those cases also return an error in case the output should be discarded as might not contain valid values (as a general Golang idiom).

Anyway, I see your point and also because of staying consistent with EKS is a value I would keep this too, this is kinda the interfacing purpose I meant originally. 👍🏻